akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRequestingRecoveryPermit.scala

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright (C) 2016-2018 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.persistence.typed.internal
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors.MutableBehavior
+import akka.actor.typed.scaladsl.{ ActorContext, Behaviors, StashBuffer, TimerScheduler }
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.persistence._
+import akka.persistence.typed.internal.EventsourcedBehavior.WriterIdentity
+
+/**
+ * INTERNAL API
+ *
+ * First (of four) behaviour of an PersistentBehaviour.
+ *
+ * Requests a permit to start recovering this actor; this is tone to avoid
+ * hammering the journal with too many concurrently recovering actors.
+ */
+@InternalApi
+private[akka] final class EventsourcedRequestingRecoveryPermit[Command, Event, State](
+  val persistenceId:    String,
+  override val context: ActorContext[Any],
```

Comment:
Seems like an anti-pattern to pass the context around all the way...

Reply:
Spent half a day today migrating away from the cake though..